### ISSUE TITLE
[BUG] FTP contents invisible

### ISSUE THREAD

**Issue by dio-lt** [PhilippC/keepass2android#2423]
### Checks

- [X] I have read the FAQ section, searched the open issues, and still think this is a new bug.

### Describe the bug you encountered:

I would like to use a database file that I am storing on my FTP storage (Router with Padavan 3.4.3.9L). If I attempt to open a file with an "FTP" storage type and enter the login data, the storage opens successfully, but it appears empty. Other software is able to see the FTP contents, including desktop KeePass 2.54.

I tried another random FTP server and I am able to see the contents, so it's just my FTP.

Here are the details that will allow you to reproduce the issue:

Filepath: ftp://5.20.218.22/H34/Support/sample.kdbx
User: support
Pass: abc123

Samsung Galaxy S22+ on Android 13. Tried both the latest Keepass2Android release and beta versions, but the behavior is the same.

![photo_2023-09-20_12-45-58](https://github.com/PhilippC/keepass2android/assets/145553088/b991e2d8-6754-45b6-8f1f-f985bf960e34)
![photo_2023-09-20_12-45-58 (2)](https://github.com/PhilippC/keepass2android/assets/145553088/ada22224-baa6-4fce-b11a-a7cee40126eb)
![filezilla](https://github.com/PhilippC/keepass2android/assets/145553088/693823fb-afbc-42a8-94eb-07f1a5946082)

### Describe what you expected to happen:

I am able to access a database on my FTP.

### What version of Keepass2Android are you using?

1.09e-r7

### Which version of Android are you on?

13

**Comment by hyproman** [comment]
I was able to reproduce this using your test server account (thanks for that; I think it would be been impossible to figure this out otherwise!). It's a strange one; I could not duplicate the issue on another FTP server, but I did confirm that it could be reproduced on your server, using a different client (ncftp). I'm honestly not sure what to make of it. I'm curious what software/version of FTP server you are running (I could not reproduce with vsftpd on Ubuntu 22.04).

Anyway, PR #2434 seems to have fixed it up. You'll just need to wait for that to be merged into the mainline, and then a new release built (unfortunately those things are out of my control).

If you're feeling adventurous and want to try out the debug build that includes this fix, the APK can be found [here](https://github.com/PhilippC/keepass2android/suites/16991300345/artifacts/970593275). Keep in mind that it is built from **master**, which may or may not be as stable as the current release...
